pkg/controller/jobframework/setup.go

```diff
@@ -94,6 +108,39 @@ func SetupControllers(mgr ctrl.Manager, log logr.Logger, opts ...Option) error {
 	})
 }
 
+func setupComponents(mgr ctrl.Manager, log logr.Logger, gvk schema.GroupVersionKind, fwkNamePrefix string, cb IntegrationCallbacks, opts ...Option) {
+	// Attempt to get the REST mapping for the GVK
+	if _, err := mgr.GetRESTMapper().RESTMapping(gvk.GroupKind(), gvk.Version); err != nil {
+		if !meta.IsNoMatchError(err) {
+			log.Error(err, fmt.Sprintf("%s: unable to get REST mapping", fwkNamePrefix))
+			return
+		}
+		log.Info("No matching API in the server for job framework, skipped setup of controller and webhook")
+	} else {
+		if err := setupControllerAndWebhook(mgr, gvk, fwkNamePrefix, cb, opts...); err != nil {
+			log.Error(err, "Failed to set up controller and webhook")
+		} else {
+			log.Info("Set up controller and webhook for job framework")
+		}
+	}
+}
+
```

Comment:
Just a nit for easier readability. Its your choice, no particular concerns from my end :)

Also, we can exit if the controller setup fails. 

```suggestion
	// Attempt to get the REST mapping for the GVK
	if _, err := mgr.GetRESTMapper().RESTMapping(gvk.GroupKind(), gvk.Version); err != nil {
    if meta.IsNoMatchError(err) {
        log.Info("No matching API in the server for job framework, skipped setup of controller and webhook")
        return
    }
    log.Error(err, fmt.Sprintf("%s: unable to get REST mapping", fwkNamePrefix))
    return
}

// If APIs are available set up the controller and web hook
if err := setupControllerAndWebhook(mgr, gvk, fwkNamePrefix, cb, opts...); err != nil {
    log.Error(err, "Failed to set up controller and webhook")
    os.Exit(1) -->  We usually exit if controller setup fails
}     
log.Info("Set up controller and webhook for job framework")
}

```